Github Action for Keeping MLST's Schemes current

erinyoung opened the pull request:
I have created a github action to keep MLST current with pubmlst.

This file is similar to that of the Github Action for Freyja.

There are some notable differences:
1. All the github action functions that are used are current.
2. There is no database version obtained from the built image. This means the tag is just `${version}-${date of deploy}`.
3. The frequency of the github action is set to every Monday at 7:30 am (as opposed to every day). I'm open to adjusting this to more or less frequently.
4. The 2.23.0 version of the dockerfile is not used. The 2.23.0-2024-01 Dockerfile is used because of how it downloads the databases.

kapsakcj commented:
amazing, thank you Kutluhan!

I would love to add that genome as a test in the dockerfile. When I get a chance I can submit a PR for a new dockerfile

erinyoung commented:
Sounds like a plan! I'll wait to update this file until the new Dockerfile is created.

kapsakcj commented:
I think I'll add the dockerfile here: https://github.com/StaPH-B/docker-builds/pull/881

so that we can test the update github actions workflows

erinyoung commented:
I noticed that https://github.com/StaPH-B/docker-builds/actions/runs/8208199568 was still giving depreciation warnings, so I've updated docker/metadata-action to v5 in build-to-deploy.yml.

erinyoung commented:
Since https://github.com/StaPH-B/docker-builds/pull/881 is merged, I've updated the yml file to correspond with the version of that file.

I've also adjusted the cron job to run at 7:30 am on the 1st of every Month.

kapsakcj commented:
Do we need to merge the PR in order to test the new workflow? or can we test it while it's on a dev branch?

erinyoung commented:
I don't know how to test it on the dev branch. My apologies.

kapsakcj commented:
No worries, I'll merge now
